File: python/6 kyu/122.py
# https://www.codewars.com/kata/5266876b8f4bf2da9b000362/train/python

# Passed

def likes(names):
    if len(names) == 0:
        prefix = "no one"
    elif len(names) == 1:
        prefix = names[0]
    elif len(names) == 2:
        prefix = " and ".join(names)
    else:
        prefix = ", ".join(names[:2]) + f" and {names[-1] if len(names) == 3 else (str(len(names) - 2) + ' others')}"

    result = prefix + (" likes this" if len(names) in [0, 1] else " like this")
    return result

output = likes(['Alex', 'Jacob', 'Mark', 'Max'])
print(output)

File: nodejs/6 kyu/122.js
// https://www.codewars.com/kata/5832514f64a4cecd1c000013/train/javascript

// Passed

function piecesValue(arr, s) {
  const hash = Object.freeze({
    queen: 9,
    rook: 5,
    bishop: 3,
    knight: 3,
    pawn: 1,
    king: 0,
  });

  let pieceColour = s[0];
  let result = arr
    .map((row) =>
      row
        .map((square) =>
          square.startsWith(pieceColour) ? hash[square.split("-")[1]] : 0
        )
        .reduce((prev, curr) => prev + curr, 0)
    )
    .reduce((prev, curr) => prev + curr, 0);
  return result;
}

const output = piecesValue(
  [
    ["b-bishop", " ", " ", " ", " ", " ", " ", " "],
    [" ", " ", "b-queen", " ", " ", " ", " ", "w-queen"],
    [" ", "b-king", " ", "b-pawn", "w-rook", " ", " ", " "],
    [" ", " ", " ", " ", "w-pawn", " ", " ", " "],
    [" ", " ", " ", " ", " ", "w-bishop", " ", " "],
    ["w-king", " ", " ", " ", " ", " ", " ", " "],
    [" ", " ", "b-pawn", "b-pawn", " ", " ", " ", " "],
    [" ", " ", " ", " ", " ", " ", " ", " "],
  ],
  "white"
);
console.log(output);